### PR TITLE
(PDB-4731) Validate extract clause filters

### PR DESCRIPTION
**Pull request by oanatmaria**

### CODE REVIEW

**Comment by stelcodes** [comment]
Just wanted to drop a suggestion, I think `(remove #(or (nil? %) (empty? %)) ...` could be replaced with `(keep seq ...`

**Review by stelcodes — COMMENTED** [review]
After talking with Austin, it seems like we could validate the extract filters by sending them back through the core.match validation process. I'm not sure what this would look like but I'm going to try to get a working example of this. I think the test could be improved by iterating over the queries and also including good queries that should return a 200 status.

File: test/puppetlabs/puppetdb/http/reports_test.clj [hunk 1]
```diff
@@ -457,6 +457,24 @@
...
+                    body))
+    (is (= 400 status)))
+
+  (let [{:keys [status body]} (query-response method endpoint ["extract" [["function" "count"] "certname"]
+                                                               ["null?" "type" false]
+                                                               ["some_invalid_string"]])]
+    (is (re-matches #".* is not a valid expression for \"extract\".*"
+                    body))
+    (is (= 400 status))))
+
```
[the remark this answers is not in the document]
Comment (oanatmaria) [reply test/puppetlabs/puppetdb/http/reports_test.clj]: Great suggestion, done!

**Review by austb — APPROVED** [review]

**Comment by austb** [comment]
I noticed the jenkins job failed, it looks transient, so I'm going to merge this, but we should keep an eye on it to make sure it gets through promotions

**Pull request merged**